README.md

```diff
@@ -58,32 +123,300 @@ client = Redis.new(semian: {
+
+#### In depth analysis of real world example
+
+If a single resource is slow every single request is going to suffer. We saw
+this in the example before. Let's illustrate this more clearly in the following
+Rails example where the user session is stored in Redis:
+
+```ruby
+def index
+  @user = fetch_user
+  @posts = Post.all
+end
+
+private
+def fetch_user
+  user = User.find(session[:user_id])
+rescue Redis::CannotConnectError
+  nil
+end
+```
+
+Our code is resilient to a failure of the session layer, it doesn't `HTTP 500`
+if the session store is unavailable (this can be tested with
+[Toxiproxy][toxiproxy]). It'll still send a `500` back if the data store where
+users and posts are stored is unavailable, but we accept that (this could be
+prevented with a caching tier or something else out of scope). 
+
+However, this code has two flaws:
+
+1. **What happens if the session storage is consistently slow?** I.e. the majority
+   of requests take, say, more than half the timeout time (but it should only
+   take `~1ms`)?
+2. **What happens if the session storage is unavailable and is not responding at
+   all?** I.e. we hit timeouts on every request.
+
+These two problems in turn have two related problems associated with them:
+request time and capacity.
+
+#### Response time
+
+Requests that attempt to access session storage are all gracefully handled, the
+`@user` will simply be `nil` which the code copes with. However, this has a
+major impact on users as every request to the session data store has to time
+out. This causes the average request time to all pages that access the session
+layer to go up by however long your timeout is. Your timeout is proportional to
+your worst case timeout, as well as the number of attempts to hit it on each
+page. This is the problem Semian solves by using heurestics to fail these
+requests early which causes a much better user experience during those times.
+
+#### Capacity loss
+
+When your single-threaded workers is waiting for a resource to return, it's
+effectively doing nothing when it could be serving fast requests. To use the
+example from before, perhaps some actions do not access the session storage at
+all. These requests will pile up behind the now slow requests that are trying to
+access that layer, because they're failing slowly. Essentially, your capacity
+degrades significantly because your average response time goes up (as explained
+in the previous section). Capacity loss is simply follows form an increase in
+response time. The higher your timeout and the slower your resource, the more
+capacity you lose.
+
+#### Timeouts aren't enough
+
+As demonstrated in the Response Time and Capacity Loss sections, timeouts aren't
+enough. Consistent timeouts will increase the average response time which causes
+a bad user experience, and ultimately compromise the performance of the entire
+system. Even if the timeout is as low as ~250ms (just enough to allow a single
```

Comment:
before, you used `markdown` for duration values. here you don't. lets make it consistent (I would say always just plain, no `markdown`)